src/server/frontend_wayland/generator/wrapper_generator.cpp

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright © 2017 Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 or 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored By: Christopher James Halse Rogers <christopher.halse.rogers@canonical.com>
+ */
+
+#include "emitter.h"
+#include "interface.h"
+#include "utils.h"
+
+#include <libxml++/libxml++.h>
+#include <iostream>
+
+Emitter comment_header(std::string const& input_file_path)
+{
+    return Lines{
+        "/*",
+        " * AUTOGENERATED - DO NOT EDIT",
+        " *",
+        {" * This header is generated by wrapper_generator.cpp from ", file_name_from_path(input_file_path)},
+        " * To regenerate, run the “refresh-wayland-wrapper” target.",
```

Comment:
In the follow-up, you can probably remove this line; the generated header will be automatically generated when needed.

Reply:
I have rebased out the cmake changes to auto generate on build. Even when it does autogenerate, it may still be useful to indicate how to manually generate them (else the target to build would be completely undocumented).